### ISSUE TITLE
Issue in tfa.metrics.F1Score

### ISSUE THREAD

**Issue by Wideeeee** [tensorflow/tensorflow#59283]
------------------------

### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: Yes
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**: - 
-   **TensorFlow installed from (source or binary)**: source
-   **TensorFlow version (use command below)**: 2.10
-   **Python version**: 3.9
-   **Bazel version (if compiling from source)**: -
-   **GCC/Compiler version (if compiling from source)**: -
-   **CUDA/cuDNN version**: -
-   **GPU model and memory**: - 
-   **Exact command to reproduce**: -

### Describe the problem

In the documentation of the tensorflow addons F1Score is not mentioned, that the metric is not native working for binary-classification. An example would be realy helpfull to understand that for the binary classification the "num_classes" argument must be set to "1" and there must be a threshold of "0.5". 

### Source code / logs

I think the following code example would clearify the use of the F1Score function for binary classification. 

f1 = tfa.metrics.F1Score(num_classes=1, average=None, threshold=0.5)
n_output = 1
model = Sequential([
    InputLayer(input_shape=(X_train.shape[1], X_train.shape[2]), name="Input"),
    Normalization(),
    Conv1D(16, kernel_size=(5), padding='same', activation='tanh'),
    BatchNormalization(),
    MaxPooling1D(),
    Conv1D(32, kernel_size=(5), padding='same', activation='tanh'),
    BatchNormalization(),
    MaxPooling1D(),
    Flatten(),
    Dropout(0.65),
    Dense(8, activation='tanh'),
    Dense(4, activation='tanh'),
    Dense(1 ,  activation="sigmoid")
])

model.compile(loss='binary_crossentropy', optimizer=Adam(learning_rate=0.001), metrics=['accuracy',f1])
model.summary()

**Comment by sachinprasadhs** [comment]
Since `tfa.metrics.F1Score` supports multi class, multi label inputs along with binary class, we have generic example defined in the documentation.

does it make sense for you if the below snippet is included for the binary classification with 1 class.

```
metric = tfa.metrics.F1Score(num_classes=1, threshold=0.5)
y_true = np.array([[1],
                   [0],
                   [1]], np.int32)
y_pred = np.array([[0.2],
                   [0.2],
                   [0.6]], np.float32)
metric.update_state(y_true, y_pred)
result = metric.result()
result.numpy()
```

**Comment by Wideeeee** [comment]
Thank you very much for the quick handling of my problem. 
I think this example would be very helpful in the documentation. 
Additionally, in the "num_classes" definition, a note would be nice that it does not correspond to the number of classes in a binary classification, but must be set to "1".
